core/frontend/src/ViewState.ts

```diff
@@ -443,10 +449,19 @@ export abstract class ViewState extends ElementState {
   }
 
   /** @internal */
-  public static getStandardViewMatrix(id: StandardViewId): Matrix3d { return StandardView.getStandardRotation(id); }
+  public static getStandardViewMatrix(id: StandardViewId): Matrix3d {
+    return StandardView.getStandardRotation(id);
+  }
 
   /** Orient this view to one of the [[StandardView]] rotations. */
-  public setStandardRotation(id: StandardViewId) { this.setRotation(ViewState.getStandardViewMatrix(id)); }
+  public setStandardRotation(id: StandardViewId) {
+    const worldToView = ViewState.getStandardViewMatrix(id);
+    const globeToWorld = this.getGlobeRotation();
+    if (globeToWorld)

```

Comment:
Seems like it would be better to add a new method rather than changing setStandardRotation?

Reply:
Done.